<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="caitlinpage" />

<meta name="date" content="2024-09-04" />

<title>fp-test</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">damsel_paper</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/caitlinpage/damsel_paper">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">fp-test</h1>
<h4 class="author">caitlinpage</h4>
<h4 class="date">2024-09-04</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-10-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>damsel_paper/analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240816code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240816)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240816code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240816)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomcaitlinpagedamselpapertree73f7dc1d28988e1706bdfa5b6a5541c1f69be327targetblank73f7dc1a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/caitlinpage/damsel_paper/tree/73f7dc1d28988e1706bdfa5b6a5541c1f69be327" target="_blank">73f7dc1</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomcaitlinpagedamselpapertree73f7dc1d28988e1706bdfa5b6a5541c1f69be327targetblank73f7dc1a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/caitlinpage/damsel_paper/tree/73f7dc1d28988e1706bdfa5b6a5541c1f69be327" target="_blank">73f7dc1</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    renv/library/
    Ignored:    renv/staging/

Untracked files:
    Untracked:  data/damsel_counts.rds
    Untracked:  data/damsel_counts_a.rds
    Untracked:  data/damsel_counts_b.rds
    Untracked:  data/sd_1_SRR794884-vs-Dam.kde-norm.gatc-FDR0.01.peaks.gff
    Untracked:  data/sd_2_SRR7948877-vs-Dam.kde-norm.gatc-FDR0.01.peaks.gff
    Untracked:  data/txdb_genes.rds
    Untracked:  output/damsel_dm.rds
    Untracked:  output/damsel_genes.rds
    Untracked:  output/damsel_peaks.rds
    Untracked:  output/fp_vissers_peaks.txt
    Untracked:  output/keep
    Untracked:  output/lrt_sd.txt
    Untracked:  output/marshall_peaks.rds
    Untracked:  output/peaks.txt
    Untracked:  output/vissers_dm.rds
    Untracked:  output/vissers_peaks.rds
    Untracked:  output/vissers_peaks.txt
    Untracked:  output/vissers_peaks_mod.rds
    Untracked:  renv/.gitignore

Unstaged changes:
    Modified:   code/run-Damsel.R
    Modified:   code/run-Vissers.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/fp-test.Rmd</code>) and HTML
(<code>docs/fp-test.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/damsel_paper/blob/73f7dc1d28988e1706bdfa5b6a5541c1f69be327/analysis/fp-test.Rmd" target="_blank">73f7dc1</a>
</td>
<td>
caitlinpage
</td>
<td>
2024-10-01
</td>
<td>
wflow_publish(c("analysis/compare_dm.Rmd", "analysis/compare_peaks.Rmd",
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/caitlinpage/damsel_paper/blob/08944f855b1bf74575277f2ac768ae846ded73b7/analysis/fp-test.Rmd" target="_blank">08944f8</a>
</td>
<td>
caitlinpage
</td>
<td>
2024-10-01
</td>
<td>
initial commit analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<pre class="r"><code>library(Damsel)
library(plyranges)</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, setdiff, table,
    tapply, union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Warning: package &#39;IRanges&#39; was built under R version 4.4.1</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>Warning: package &#39;S4Vectors&#39; was built under R version 4.4.1</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:utils&#39;:

    findMatches</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    expand.grid, I, unname</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>
Attaching package: &#39;plyranges&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:IRanges&#39;:

    slice</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    filter</code></pre>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:plyranges&#39;:

    between, n, n_distinct</code></pre>
<pre><code>The following objects are masked from &#39;package:GenomicRanges&#39;:

    intersect, setdiff, union</code></pre>
<pre><code>The following object is masked from &#39;package:GenomeInfoDb&#39;:

    intersect</code></pre>
<pre><code>The following objects are masked from &#39;package:IRanges&#39;:

    collapse, desc, intersect, setdiff, slice, union</code></pre>
<pre><code>The following objects are masked from &#39;package:S4Vectors&#39;:

    first, intersect, rename, setdiff, setequal, union</code></pre>
<pre><code>The following objects are masked from &#39;package:BiocGenerics&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(ggplot2)
library(BSgenome.Dmelanogaster.UCSC.dm6)</code></pre>
<pre><code>Loading required package: BSgenome</code></pre>
<pre><code>Loading required package: Biostrings</code></pre>
<pre><code>Loading required package: XVector</code></pre>
<pre><code>
Attaching package: &#39;Biostrings&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    strsplit</code></pre>
<pre><code>Loading required package: BiocIO</code></pre>
<pre><code>Loading required package: rtracklayer</code></pre>
<pre><code>
Attaching package: &#39;rtracklayer&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:BiocIO&#39;:

    FileForFormat</code></pre>
<pre class="r"><code>library(edgeR)</code></pre>
<pre><code>Warning: package &#39;edgeR&#39; was built under R version 4.4.1</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre><code>Warning: package &#39;limma&#39; was built under R version 4.4.1</code></pre>
<pre><code>
Attaching package: &#39;limma&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:BiocGenerics&#39;:

    plotMA</code></pre>
<pre class="r"><code>test_vissers_dm &lt;- function(damsel_counts) {
  matrix &lt;- as.matrix(damsel_counts[, grepl(&quot;bam&quot;, colnames(damsel_counts), ignore.case = TRUE)])
  rownames(matrix) &lt;- damsel_counts$Position
  group = c(&quot;Dam&quot;, &quot;Sd&quot;, &quot;Dam&quot;, &quot;Sd&quot;)
  design = model.matrix(~group)
  y = DGEList(matrix, group = group)
  keep &lt;- rowSums(cpm(y)&gt;=0.5) &gt;= 2
  y = y[keep, ,keep.lib.sizes=FALSE]
  y = calcNormFactors(y)
  y = estimateDisp(y, robust = T, design = design)

  fit = glmFit(y, design = design)
  lrt = glmLRT(fit, coef=2)
  de.Sd &lt;- decideTestsDGE(lrt, lfc = 1)
  lrt$table$significant &lt;- de.Sd

  vissers_dm &lt;- data.frame(lrt$table)
  vissers_dm$significant &lt;- data.frame(de.Sd)$groupSd
  vissers_dm
}
test_vissers_peaks &lt;- function(damsel_counts) {
  matrix &lt;- as.matrix(damsel_counts[, grepl(&quot;bam&quot;, colnames(damsel_counts), ignore.case = TRUE)])
  rownames(matrix) &lt;- damsel_counts$Position
  group = c(&quot;Dam&quot;, &quot;Sd&quot;, &quot;Dam&quot;, &quot;Sd&quot;)
  design = model.matrix(~group)
  y = DGEList(matrix, group = group)
  keep &lt;- rowSums(cpm(y)&gt;=0.5) &gt;= 2
  y = y[keep, ,keep.lib.sizes=FALSE]
  y = calcNormFactors(y)
  y = estimateDisp(y, robust = T, design = design)

  fit = glmFit(y, design = design)
  lrt = glmLRT(fit, coef=2)
  de.Sd &lt;- decideTestsDGE(lrt, lfc = 1)
  lrt$table$significant &lt;- de.Sd

  vissers_dm &lt;- data.frame(lrt$table)
  vissers_dm$significant &lt;- data.frame(de.Sd)$groupSd

  write.table(vissers_dm, file=&#39;../output/lrt_sd.txt&#39;, quote=F)
  write.table(keep, file=&#39;../output/keep&#39;, quote=F, col.names = FALSE)

  system2(&quot;python3&quot;, args=c(&quot;../code/call_peaks.py&quot;,
          &quot;../output/keep&quot;, &quot;../output/lrt_sd.txt&quot;, &quot;&gt;&quot;,
          &quot;../output/fp_vissers_peaks.txt&quot;))
  vissers_peaks &lt;- read.table(&quot;../output/fp_vissers_peaks.txt&quot;)

  names(vissers_peaks) &lt;- c(&#39;seqnames&#39;, &#39;start&#39;, &#39;end&#39;, &quot;tags&quot;, &#39;pen&#39;, &#39;aveLogFC&#39;, &#39;sig&#39;)
  vissers_peaks
}</code></pre>
<p>The Samples are: * D1: Dam1 * F1: Sd1 * D2: Dam2 * F2: Sd2</p>
<pre class="r"><code>damsel_counts &lt;- rbind(readRDS(&quot;../data/damsel_counts_a.rds&quot;), readRDS(&quot;../data/damsel_counts_b.rds&quot;))
gatc_regions &lt;- getGatcRegions(BSgenome.Dmelanogaster.UCSC.dm6::BSgenome.Dmelanogaster.UCSC.dm6)$regions</code></pre>
<pre><code>Warning in .local(x, row.names, optional, ...): &#39;optional&#39; argument was ignored
Warning in .local(x, row.names, optional, ...): &#39;optional&#39; argument was ignored
Warning in .local(x, row.names, optional, ...): &#39;optional&#39; argument was ignored
Warning in .local(x, row.names, optional, ...): &#39;optional&#39; argument was ignored
Warning in .local(x, row.names, optional, ...): &#39;optional&#39; argument was ignored
Warning in .local(x, row.names, optional, ...): &#39;optional&#39; argument was ignored
Warning in .local(x, row.names, optional, ...): &#39;optional&#39; argument was ignored</code></pre>
<pre><code>Warning in GenomeInfoDb::renameSeqlevels(x = df_, value = newStyle): invalid
seqlevels &#39;chrM&#39; ignored</code></pre>
<pre class="r"><code>head(damsel_counts)</code></pre>
<pre><code>           Position seqnames start end width strand dam_1_SRR7948872.BAM
chr2L-82   chr2L-82    chr2L    82 230   149      *                  1.0
chr2L-231 chr2L-231    chr2L   231 371   141      *                  1.5
chr2L-372 chr2L-372    chr2L   372 539   168      *                  2.5
chr2L-540 chr2L-540    chr2L   540 688   149      *                  2.0
chr2L-689 chr2L-689    chr2L   689 829   141      *                  0.0
chr2L-830 chr2L-830    chr2L   830 997   168      *                  0.0
          sd_1_SRR7948874.BAM dam_2_SRR7948876.BAM sd_2_SRR7948877.BAM
chr2L-82                 0.33                  0.0                 0.0
chr2L-231                5.67                 87.0                57.5
chr2L-372                6.17                 88.0                58.5
chr2L-540                4.83                  0.0                 0.0
chr2L-689                0.00                  0.5                 0.5
chr2L-830                1.33                  4.5                 3.5</code></pre>
</div>
<div id="we-can-rearrange-them-as" class="section level1">
<h1>We can rearrange them as:</h1>
<ul>
<li>D1: Dam1</li>
<li>F1: Dam2</li>
<li>D2: Sd1</li>
<li>F2: Sd2</li>
</ul>
<pre class="r"><code>damsel_counts_a &lt;- damsel_counts[,c(1:6,7,9,8,10)]
head(damsel_counts_a)</code></pre>
<pre><code>           Position seqnames start end width strand dam_1_SRR7948872.BAM
chr2L-82   chr2L-82    chr2L    82 230   149      *                  1.0
chr2L-231 chr2L-231    chr2L   231 371   141      *                  1.5
chr2L-372 chr2L-372    chr2L   372 539   168      *                  2.5
chr2L-540 chr2L-540    chr2L   540 688   149      *                  2.0
chr2L-689 chr2L-689    chr2L   689 829   141      *                  0.0
chr2L-830 chr2L-830    chr2L   830 997   168      *                  0.0
          dam_2_SRR7948876.BAM sd_1_SRR7948874.BAM sd_2_SRR7948877.BAM
chr2L-82                   0.0                0.33                 0.0
chr2L-231                 87.0                5.67                57.5
chr2L-372                 88.0                6.17                58.5
chr2L-540                  0.0                4.83                 0.0
chr2L-689                  0.5                0.00                 0.5
chr2L-830                  4.5                1.33                 3.5</code></pre>
<div id="damsel" class="section level2">
<h2>Damsel</h2>
<pre class="r"><code>damsel_fp &lt;- testDmRegions(makeDGE(damsel_counts_a, min.samples = 2), gatc_regions)</code></pre>
<pre><code>Warning in plot.xy(xy.coords(x, y), type = type, ...): &quot;panel.first&quot; is not a
graphical parameter</code></pre>
<p><img src="figure/fp-test.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>damsel_fp %&gt;% group_by(meth_status) %&gt;% summarise(n=n())</code></pre>
<pre><code># A tibble: 3 × 2
  meth_status       n
  &lt;chr&gt;         &lt;int&gt;
1 No_sig       129274
2 Not_included 232982
3 Upreg         21398</code></pre>
<pre class="r"><code>21398/(21398+129274)</code></pre>
<pre><code>[1] 0.1420171</code></pre>
<pre class="r"><code>nrow(damsel_fp)</code></pre>
<pre><code>[1] 383654</code></pre>
<pre class="r"><code>damsel_dm &lt;- readRDS(&quot;../output/damsel_dm.rds&quot;)
damsel_dm %&gt;% group_by(meth_status) %&gt;% summarise(n=n())</code></pre>
<pre><code># A tibble: 3 × 2
  meth_status       n
  &lt;chr&gt;         &lt;int&gt;
1 No_sig       113787
2 Not_included 232982
3 Upreg         36885</code></pre>
</div>
<div id="vissers" class="section level2">
<h2>Vissers</h2>
<pre class="r"><code>vissers_fp &lt;- test_vissers_dm(damsel_counts_a)</code></pre>
<pre><code>Warning: &#39;decideTestsDGE&#39; is deprecated.
Use &#39;decideTests&#39; instead.
See help(&quot;Deprecated&quot;)</code></pre>
<pre class="r"><code>vissers_fp %&gt;% group_by(significant) %&gt;% summarise(n=n())</code></pre>
<pre><code># A tibble: 3 × 2
  significant      n
        &lt;int&gt;  &lt;int&gt;
1          -1   1112
2           0 148319
3           1   1254</code></pre>
<ul>
<li><p>both identify significant results</p></li>
<li><p>we suspect this is because the significant difference in library
size between the replicates creates significance</p></li>
<li><p>check for peaks</p></li>
</ul>
<pre class="r"><code>nrow(identifyPeaks(damsel_fp))</code></pre>
<pre><code>[1] 1388</code></pre>
<pre class="r"><code>nrow(test_vissers_peaks(damsel_counts_a))</code></pre>
<pre><code>Warning: &#39;decideTestsDGE&#39; is deprecated.
Use &#39;decideTests&#39; instead.
See help(&quot;Deprecated&quot;)</code></pre>
<pre><code>[1] 186</code></pre>
<ul>
<li>186 peaks</li>
</ul>
</div>
</div>
<div id="or" class="section level1">
<h1>Or …</h1>
<ul>
<li>D1: Dam1</li>
<li>F1: Sd2</li>
<li>D2: Dam2</li>
<li>F2: Sd1</li>
</ul>
<pre class="r"><code>damsel_counts_a &lt;- damsel_counts[,c(1:6,7,10,9,8)]
head(damsel_counts_a)</code></pre>
<pre><code>           Position seqnames start end width strand dam_1_SRR7948872.BAM
chr2L-82   chr2L-82    chr2L    82 230   149      *                  1.0
chr2L-231 chr2L-231    chr2L   231 371   141      *                  1.5
chr2L-372 chr2L-372    chr2L   372 539   168      *                  2.5
chr2L-540 chr2L-540    chr2L   540 688   149      *                  2.0
chr2L-689 chr2L-689    chr2L   689 829   141      *                  0.0
chr2L-830 chr2L-830    chr2L   830 997   168      *                  0.0
          sd_2_SRR7948877.BAM dam_2_SRR7948876.BAM sd_1_SRR7948874.BAM
chr2L-82                  0.0                  0.0                0.33
chr2L-231                57.5                 87.0                5.67
chr2L-372                58.5                 88.0                6.17
chr2L-540                 0.0                  0.0                4.83
chr2L-689                 0.5                  0.5                0.00
chr2L-830                 3.5                  4.5                1.33</code></pre>
<div id="damsel-1" class="section level2">
<h2>Damsel</h2>
<pre class="r"><code>damsel_fp &lt;- testDmRegions(makeDGE(damsel_counts_a, min.samples = 2), gatc_regions)</code></pre>
<pre><code>Warning in plot.xy(xy.coords(x, y), type = type, ...): &quot;panel.first&quot; is not a
graphical parameter</code></pre>
<p><img src="figure/fp-test.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>damsel_fp %&gt;% group_by(meth_status) %&gt;% summarise(n=n())</code></pre>
<pre><code># A tibble: 2 × 2
  meth_status       n
  &lt;chr&gt;         &lt;int&gt;
1 No_sig       150672
2 Not_included 232982</code></pre>
<ul>
<li>no significance identified</li>
</ul>
</div>
<div id="vissers-1" class="section level2">
<h2>Vissers</h2>
<pre class="r"><code>vissers_fp &lt;- test_vissers_dm(damsel_counts_a)</code></pre>
<pre><code>Warning: &#39;decideTestsDGE&#39; is deprecated.
Use &#39;decideTests&#39; instead.
See help(&quot;Deprecated&quot;)</code></pre>
<pre class="r"><code>vissers_fp %&gt;% group_by(significant) %&gt;% summarise(n=n())</code></pre>
<pre><code># A tibble: 3 × 2
  significant      n
        &lt;int&gt;  &lt;int&gt;
1          -1  20346
2           0 110709
3           1  19630</code></pre>
<ul>
<li>significance identified</li>
</ul>
<pre class="r"><code>nrow(test_vissers_peaks(damsel_counts_a))</code></pre>
<pre><code>Warning: &#39;decideTestsDGE&#39; is deprecated.
Use &#39;decideTests&#39; instead.
See help(&quot;Deprecated&quot;)</code></pre>
<pre><code>[1] 2286</code></pre>
<ul>
<li>more peaks than Damsel identifies the other way around</li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R Under development (unstable) (2024-01-17 r85813)
Platform: x86_64-apple-darwin20
Running under: macOS Sonoma 14.1.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Melbourne
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices datasets  utils     methods  
[8] base     

other attached packages:
 [1] edgeR_4.2.1                           limma_3.60.4                         
 [3] BSgenome.Dmelanogaster.UCSC.dm6_1.4.1 BSgenome_1.72.0                      
 [5] rtracklayer_1.64.0                    BiocIO_1.14.0                        
 [7] Biostrings_2.72.1                     XVector_0.44.0                       
 [9] ggplot2_3.5.1                         dplyr_1.1.4                          
[11] plyranges_1.24.0                      GenomicRanges_1.56.1                 
[13] GenomeInfoDb_1.40.1                   IRanges_2.38.1                       
[15] S4Vectors_0.42.1                      BiocGenerics_0.50.0                  
[17] Damsel_1.0.0                          workflowr_1.7.1                      

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.1            bitops_1.0-8               
 [3] fastmap_1.2.0               RCurl_1.98-1.16            
 [5] GenomicAlignments_1.40.0    promises_1.3.0             
 [7] XML_3.99-0.17               digest_0.6.35              
 [9] lifecycle_1.0.4             statmod_1.5.0              
[11] processx_3.8.4              magrittr_2.0.3.9000        
[13] compiler_4.4.0              rlang_1.1.3                
[15] sass_0.4.9                  tools_4.4.0                
[17] utf8_1.2.4                  yaml_2.3.8                 
[19] knitr_1.46                  S4Arrays_1.4.1             
[21] curl_5.2.1                  DelayedArray_0.30.1        
[23] abind_1.4-5                 BiocParallel_1.38.0        
[25] purrr_1.0.2                 withr_3.0.1                
[27] grid_4.4.0                  fansi_1.0.6                
[29] git2r_0.33.0                colorspace_2.1-1           
[31] scales_1.3.0                SummarizedExperiment_1.34.0
[33] cli_3.6.2                   rmarkdown_2.27             
[35] crayon_1.5.3                generics_0.1.3             
[37] rstudioapi_0.16.0           httr_1.4.7                 
[39] rjson_0.2.21                cachem_1.1.0               
[41] stringr_1.5.1.9000          splines_4.4.0              
[43] zlibbioc_1.50.0             parallel_4.4.0             
[45] BiocManager_1.30.23         restfulr_0.0.15            
[47] matrixStats_1.3.0           vctrs_0.6.5                
[49] Matrix_1.7-0                jsonlite_1.8.8             
[51] callr_3.7.6                 locfit_1.5-9.10            
[53] tidyr_1.3.1                 jquerylib_0.1.4            
[55] glue_1.7.0                  codetools_0.2-20           
[57] ps_1.7.6                    stringi_1.8.4              
[59] gtable_0.3.5                later_1.3.2                
[61] UCSC.utils_1.0.0            munsell_0.5.1              
[63] tibble_3.2.1                pillar_1.9.0               
[65] htmltools_0.5.8.1           GenomeInfoDbData_1.2.12    
[67] R6_2.5.1                    rprojroot_2.0.4            
[69] evaluate_0.23               Biobase_2.64.0             
[71] lattice_0.22-6              highr_0.10                 
[73] Rsamtools_2.20.0            renv_1.0.7                 
[75] httpuv_1.6.15               bslib_0.7.0                
[77] Rcpp_1.0.12                 SparseArray_1.4.8          
[79] whisker_0.4.1               xfun_0.44                  
[81] fs_1.6.4                    MatrixGenerics_1.16.0      
[83] getPass_0.2-4               pkgconfig_2.0.3            </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
