---
title: "fp-test"
author: "caitlinpage"
date: "2024-09-04"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: inline
---

## Introduction

```{r}
library(Damsel)
library(plyranges)
library(dplyr)
library(ggplot2)
library(BSgenome.Dmelanogaster.UCSC.dm6)
library(edgeR)
```
```{r, echo=FALSE}
makeDGE2 <- function(counts.df, max.width=10000, lib.size=NULL, min.cpm=0.5, min.samples=3, include_replicate=TRUE, design=NULL) {
    if (!is.data.frame(counts.df)) {
        stop("Must have data.frame of counts")
    }
    if (!is.numeric(min.cpm) | length(min.cpm) > 1) {
        stop("min.cpm must be 1 value, recommend using default value")
    }
    if (!is.numeric(min.samples) | length(min.samples) > 1) {
        stop("min.samples must be 1 value, recommend using default value")
    }
    if(!"width" %in% colnames(counts.df)) {
        counts.df <- counts.df %>% dplyr::mutate(end = as.double(.data$end),
            start = as.double(.data$start))
        counts.df <- dplyr::mutate(counts.df, width = end - start + 1)
    }
    counts.df <- counts.df %>% dplyr::filter(.data$width <= max.width)
    if(length(grep("bam", colnames(counts.df), ignore.case = TRUE)) == 0) {
        stop("Samples must have bam or BAM in the name")
    }
    matrix <- as.matrix(counts.df[, grepl("bam", colnames(counts.df), ignore.case = TRUE)])
    rownames(matrix) <- counts.df$Position

    n_samples <- seq_len(ncol(matrix) / 2)

    group <- rep(c("Dam", "Fusion"), times = length(n_samples))

    dge <- edgeR::DGEList(matrix, lib.size = lib.size, group = group, gene = counts.df[, 2:5])

    keep <- rowSums(edgeR::cpm(dge) >= min.cpm) >= min.samples
    dge <- dge[keep, , keep.lib.sizes = FALSE]

    dge <- edgeR::calcNormFactors(dge)
    if(is.null(design)) {
    group <- c("Dam", "Sd", "Dam", "Sd")
    rep <- c(0,0,1,1)
   # design <- model.matrix(~group + rep)
    if(include_replicate == TRUE) {
      design <- model.matrix(~group + rep)
    } else {
      design <- model.matrix(~group)
    }
    } 
    #design
    dge <- edgeR::estimateDisp(dge, robust = TRUE, design = design)

    dge
}

```


```{r}
damsel_counts <- rbind(readRDS("../data/damsel_counts_a.rds"), readRDS("../data/damsel_counts_b.rds"))
gatc_regions <- getGatcRegions(BSgenome.Dmelanogaster.UCSC.dm6::BSgenome.Dmelanogaster.UCSC.dm6)$regions
```
Normally, the design matrix in Damsel for differential testing has 2 columns, 1 for the grouping factor (Dam, Fusion), and an optional column for the replicates.


Normal format
* A: group1 = dam1, dam2 | group2 = fusion1, fusion2

False positive testing 
* B: group1 = dam1, fusion2 | group2 = dam2, fusion1 
* C: group1 = dam1, fusion1 | group2 = dam2, fusion2

To do Damsel's false positive testing, the replicates will not be included.

# A - Normal analysis

The Samples are:
* D1: Dam1
* F1: Sd1
* D2: Dam2
* F2: Sd2

```{r}
head(damsel_counts)
```
```{r}
testDmRegions(makeDGE2(damsel_counts, include_replicate = FALSE), regions = gatc_regions) %>% 
  group_by(meth_status) %>%
  summarise(n=n())

testDmRegions(makeDGE2(damsel_counts, include_replicate = TRUE), regions = gatc_regions) %>% 
  group_by(meth_status) %>%
  summarise(n=n())
```

```{r shared-code4}
source("../code/helper-vissers.R")
test_vissers_dm(damsel_counts)$design
test_vissers_dm(damsel_counts)$results %>% 
  group_by(significant) %>%
  summarise(n=n())
```


# B: False positive test
* group1 = dam1, fusion2 | group2 = dam2, fusion1 
```{r}
head(damsel_counts)
```
* rearrange samples to reflect new groups
```{r}
damsel_counts_a <- damsel_counts[,c(1:7,9,10,8)]
head(damsel_counts_a)
```

Test Damsel

```{r}
makeDGE2(damsel_counts_a, min.samples = 2, include_replicate=FALSE)$design
testDmRegions(makeDGE2(damsel_counts_a, min.samples = 2, include_replicate=FALSE), gatc_regions) %>% 
  group_by(meth_status) %>% summarise(n=n())
```

Test Vissers
```{r shared-code}
source("../code/helper-vissers.R")
test_vissers_dm(damsel_counts_a)$design
test_vissers_dm(damsel_counts_a)$results %>% group_by(significant) %>% summarise(n=n())
```

# C - false positive test
* C: group1 = dam1, fusion1 | group2 = dam2, fusion2

```{r}
head(damsel_counts)
```
* rearrange samples to reflect new groups
```{r}
damsel_counts_a <- damsel_counts[,c(1:7,9,8,10)]
head(damsel_counts_a)
```

Test Damsel

```{r}
makeDGE2(damsel_counts_a, min.samples = 2, include_replicate=FALSE)$design
testDmRegions(makeDGE2(damsel_counts_a, min.samples = 2, include_replicate=FALSE), gatc_regions) %>% 
  group_by(meth_status) %>% summarise(n=n())
```

Test Vissers
```{r shared-code2}
source("../code/helper-vissers.R")
test_vissers_dm(damsel_counts_a)$results %>% group_by(significant) %>% summarise(n=n())
```
```{r shared-code3}
source("../code/helper-vissers.R")
test_vissers_peaks(damsel_counts_a)
```



