---
title: "run-Damsel"
author: "caitlinpage"
date: "2024-08-16"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: inline
---

## Introduction

```{r}
#renv::install("bioc::Damsel")
library(Damsel)
library(plyranges)
library(dplyr)
library(ggplot2)

library(BSgenome.Dmelanogaster.UCSC.dm6)
library(TxDb.Dmelanogaster.UCSC.dm6.ensGene)
library(org.Dm.eg.db)
```
```{r}
```{r, echo=FALSE}
makeDGE2 <- function(counts.df, max.width=10000, lib.size=NULL, min.cpm=0.5, min.samples=3, include_replicate=TRUE, group=NULL, design=NULL) {
    if (!is.data.frame(counts.df)) {
        stop("Must have data.frame of counts")
    }
    if (!is.numeric(min.cpm) | length(min.cpm) > 1) {
        stop("min.cpm must be 1 value, recommend using default value")
    }
    if (!is.numeric(min.samples) | length(min.samples) > 1) {
        stop("min.samples must be 1 value, recommend using default value")
    }
    if(!"width" %in% colnames(counts.df)) {
        counts.df <- counts.df %>% dplyr::mutate(end = as.double(.data$end),
            start = as.double(.data$start))
        counts.df <- dplyr::mutate(counts.df, width = end - start + 1)
    }
    counts.df <- counts.df %>% dplyr::filter(.data$width <= max.width)
    if(length(grep("bam", colnames(counts.df), ignore.case = TRUE)) == 0) {
        stop("Samples must have bam or BAM in the name")
    }
    matrix <- as.matrix(counts.df[, grepl("bam", colnames(counts.df), ignore.case = TRUE)])
    rownames(matrix) <- counts.df$Position

    n_samples <- seq_len(ncol(matrix) / 2)
    if(is.null(group)) {
        group <- rep(c("Dam", "Fusion"), times = length(n_samples))
    } else {
        group <- group
    }

    dge <- edgeR::DGEList(matrix, lib.size = lib.size, group = group, gene = counts.df[, 2:5])

    keep <- rowSums(edgeR::cpm(dge) >= min.cpm) >= min.samples
    dge <- dge[keep, , keep.lib.sizes = FALSE]

    dge <- edgeR::calcNormFactors(dge)
    if(is.null(design)) {
    rep <- c(0,0,1,1)
   # design <- model.matrix(~group + rep)
    if(include_replicate == TRUE) {
      design <- model.matrix(~group + rep)
    } else {
      design <- model.matrix(~group)
    }
    } 
    #design
    dge <- edgeR::estimateDisp(dge, robust = TRUE, design = design)

    dge
}

```
```

```{r}
gatc_regions_and_sites <- getGatcRegions(BSgenome.Dmelanogaster.UCSC.dm6::BSgenome.Dmelanogaster.UCSC.dm6)
gatc_regions <- gatc_regions_and_sites$regions
```


```{r}
damsel_counts <- rbind(readRDS("../data/damsel_counts_a.rds"), readRDS("../data/damsel_counts_b.rds"))
head(damsel_counts)
```

```{r}
damsel_dge <- makeDGE2(damsel_counts, min.samples = 2)
damsel_dge
```
```{r}
damsel_dm <- testDmRegions(damsel_dge, gatc_regions)
head(damsel_dm)
```

```{r}
damsel_peaks <- identifyPeaks(damsel_dm)
head(damsel_peaks)
nrow(damsel_peaks)
```
```{r}
txdb <- TxDb.Dmelanogaster.UCSC.dm6.ensGene
genes <- collateGenes(genes = txdb, regions = gatc_regions, org.Db = org.Dm.eg.db)
genes
```

```{r}
damsel_genes <- annotatePeaksGenes(damsel_peaks, genes, gatc_regions)
```


# Figure 1
```{r}
damsel_peaks[1,]
```

```{r}
plotCounts(damsel_counts,
     seqnames = "chr2R",
     start_region = 23076454-500,
     end_region = 23079441+500,
     log2_scale = FALSE
 ) +
  geom_dm(damsel_dm) +
  geom_peak(damsel_peaks) +
  geom_gatc(gatc_regions_and_sites$sites) +
  geom_genes_tx(genes, txdb)
```


