---
title: "run-Damsel"
author: "caitlinpage"
date: "2024-08-16"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: inline
---

## Introduction

```{r}
#renv::install("bioc::Damsel")
library(Damsel)
library(plyranges)
library(dplyr)
library(ggplot2)

library(BSgenome.Dmelanogaster.UCSC.dm6)
library(TxDb.Dmelanogaster.UCSC.dm6.ensGene)
library(org.Dm.eg.db)
```

```{r}
gatc_regions_and_sites <- getGatcRegions(BSgenome.Dmelanogaster.UCSC.dm6::BSgenome.Dmelanogaster.UCSC.dm6)
gatc_regions <- gatc_regions_and_sites$regions
```


```{r}
damsel_counts <- rbind(readRDS("../data/damsel_counts_a.rds"), readRDS("../data/damsel_counts_b.rds"))
head(damsel_counts)
```

```{r}
damsel_dge <- makeDGE(damsel_counts, min.samples = 2)
damsel_dge
```
```{r}
matrix <- as.matrix(damsel_counts[, grepl("bam", colnames(damsel_counts), ignore.case = TRUE)])
rownames(matrix) <- damsel_counts$Position

n_samples <- seq_len(ncol(matrix) / 2)

group <- rep(c("Dam", "Fusion"), times = length(n_samples))
design_df <- seq_len(ncol(matrix)) %>%
        data.frame() %>%
        stats::setNames("group")
    zero_vec <- rep(0, times = length(n_samples))
    for (i in n_samples) {
        design <- replace(zero_vec, i, 1)
        design <- rep(design, each = 2)
        design_df[, ncol(design_df) + 1] <- design
    }
design <- stats::model.matrix(~., data = design_df[, seq_len(ncol(design_df)) - 1])
```
```{r}
setNames(design_df, c("num", "group", "replicate"))
model.matrix(~., data=design_df[,2:3])
```
```{r}
group <- c("Dam", "Sd", "Dam", "Sd")
rep <- c(0,0,1,1)
group
model.matrix(~group + rep)
```

```{r}
stats::model.matrix(~., data = design_df[, seq_len(ncol(design_df))])
```

```{r}
damsel_dm <- testDmRegions(damsel_dge, gatc_regions)
head(damsel_dm)
```

```{r}
damsel_peaks <- identifyPeaks(damsel_dm)
head(damsel_peaks)
nrow(damsel_peaks)
```
```{r}
txdb <- TxDb.Dmelanogaster.UCSC.dm6.ensGene
genes <- collateGenes(genes = txdb, regions = gatc_regions, org.Db = org.Dm.eg.db)
genes
```

```{r}
damsel_genes <- annotatePeaksGenes(damsel_peaks, genes, gatc_regions)
```


# Figure 1
```{r}
damsel_peaks[1,]
```

```{r}
plotCounts(damsel_counts,
     seqnames = "chr2R",
     start_region = 23076454-500,
     end_region = 23079441+500,
     log2_scale = FALSE
 ) +
  geom_dm(damsel_dm) +
  geom_peak(damsel_peaks) +
  geom_gatc(gatc_regions_and_sites$sites) +
  geom_genes_tx(genes, txdb)
```


